test/gtest/uct/ib/test_sockaddr.cc

```diff
@@ -110,9 +110,12 @@ class test_uct_sockaddr : public uct_test {
         size_t *max_conn_priv = (size_t*)arg;
         size_t priv_data_len;
 
-        priv_data_len = uct_test::entity::priv_data_do_pack(priv_data);
-        EXPECT_LE(priv_data_len, (*max_conn_priv));
+        priv_data_len = 1 + uct_test::entity::client_priv_data.length();
+
+        memcpy(priv_data, uct_test::entity::client_priv_data.c_str(),
```

Comment:
maybe use strcpy everywhere for strings? then can avoid `\0` handling like does `client_cm_priv_data_cb`

Reply:
@evgeny-leksikov you mean use strcpy in client_cm_priv_data_cb instead of strncpy?

Reply:
@alinask I mean use strcpy instead of memcpy. result of strcpy is null-terminated string

Reply:
but in client_cm_priv_data_cb, there is a copy of a struct and strncpy is used for the str.

Reply:
If you do `strcpy` here and send `\0` in private data you can do `strcpy` or build `std::string` (if needed) on server side.
regarding to `client_cm_priv_data_cb`, why `c_data.priv_str[sizeof(c_data.priv_str) -1] = '\0';` is needed if you use `strncpy`?

Reply:
@evgeny-leksikov so you do mean to change memcpy in here and not in client_cm_priv_data_cb. 

the line with '\0' is needed because if using only:
`strncpy(c_data.priv_str, entity::client_priv_data.c_str(), sizeof(c_data.priv_str));`
it works but coverity complains - 
"Calling strncpy with a maximum size argument of 32 bytes on destination array c_data.priv_str of size 32 bytes might leave the destination string unterminated."

Reply:
Also, in client_iface_priv_data_cb, since priv_data is (void*), don't you think that memcpy is more suited here?

Reply:
> @evgeny-leksikov so you do mean to change memcpy in here and not in client_cm_priv_data_cb.
> 
> the line with '\0' is needed because if using only:
> `strncpy(c_data.priv_str, entity::client_priv_data.c_str(), sizeof(c_data.priv_str));`
> it works but coverity complains -
> "Calling strncpy with a maximum size argument of 32 bytes on destination array c_data.priv_str of size 32 bytes might leave the destination string unterminated."

Ok. then maybe use `strcpy` and add an assertion `entity::client_priv_data.length() < sizeof(c_data.priv_str)` before? IMO, it's look clearer.

Reply:
> Also, in client_iface_priv_data_cb, since priv_data is (void*), don't you think that memcpy is more suited here?

Since you copy str with `\0`, `memcpy` is fine